src/benchmarkstt/metrics/core.py

```diff
@@ -159,6 +160,157 @@ def compare(self, ref: Schema, hyp: Schema) -> OpcodeCounts:
+
+    # find the position of one entity
+    # an entity can contain more than one word
+    @staticmethod
+    def __find_pattern(search_list, complex_entity):
+        entity = ''.join(complex_entity).split(' ')
+        le = len(entity)
+        # complex_entity = [entity1 entity2 ...]
+        # the cursor sweep complex_entity to find consecutive entities entity1 ... entity2
+        cursor = 0
+        idx_found = []
+        for idx, elt in enumerate(search_list):
+            if elt == entity[cursor]:
+                cursor += 1
+                if cursor == le:
+                    idx_found.append([idx for idx in range(idx - le + 1, idx - le + 1 + le)])
+                    cursor = 0
+            else:
+                cursor = 0
+        return idx_found
+
+    # generate a list containing the detected entities in list_parsed
+    @staticmethod
+    def __generate_list_entity(self, list_parsed):
+
+        list_entity = []
+        index_entities = []
+        entities = self._entities
+        for entity in entities:
+            index_entity = self.__find_pattern(list_parsed, entity)
+            index_entities.extend(index_entity)
+
+        # sort on the position of the first part of the entity
+        index_entities.sort(key=lambda l: l[0])
+
+        # copy-past the entity found in the list
+        for k_list in index_entities:
+            list_entity.append(' '.join(list_parsed[k_list[0]:k_list[-1] + 1]))
+
+        return list_entity
+
+    # computes the BEER
+    @staticmethod
+    def compute_beer(self, list_hypothesis_entity, list_reference_entity):
+        beer = {}
+        beer_av = 0
+        entities = self._entities
+        for idx, entity in enumerate(entities):
+            count_hypothesis = list_hypothesis_entity.count(entity)
+            count_ref = list_reference_entity.count(entity)
+            beer_entity = 0
+            if count_ref != 0:
+                beer_entity = round(abs(count_ref - count_hypothesis) / count_ref, 3)
+                # accumulate the distance per entity
+                beer_av += abs(count_ref - count_hypothesis) * self._weight[idx]
+            beer[entity] = {'beer': beer_entity, 'occurrence_ref': count_ref}
+
+        l_ref = len(list_reference_entity)
+        if l_ref > 0:
+            beer_av = round(beer_av / l_ref, 3)
+        else:
+            beer_av = 0
+        beer['w_av_beer'] = {'beer': beer_av, 'occurrence_ref': l_ref}
+        return beer
+
+    def compare(self, ref: Schema, hyp: Schema):
+        # get the list of reference and hypothesis
+        ref_list = [i['item'] for i in ref]
+        hyp_list = [i['item'] for i in hyp]
+
+        # extract the entities
+        list_hypothesis_entity = self.__generate_list_entity(self, hyp_list)
+        list_reference_entity = self.__generate_list_entity(self, ref_list)
+        # compute the score
+        wer_entity = self.compute_beer(self, list_hypothesis_entity, list_reference_entity)
```

Comment:
Since we can change `__generate_list_entity` and `compute_beer` so that they aren't `@staticmethod`, you can remove the `self` from the function call's parameters.

```suggestion
        list_hypothesis_entity = self.__generate_list_entity(hyp_list)
        list_reference_entity = self.__generate_list_entity(ref_list)
        # compute the score
        wer_entity = self.compute_beer(list_hypothesis_entity, list_reference_entity)
```

Reply:
ok done